### PR TITLE
Update feed

### LINKED ISSUE #42: changes for /name/feed
1. To be compliant with the Atom Syndication Format /name/feed/ Atom Feed should have author field at least at feed level.
2. For `<link>` in `<entry>`, if there is a port for the site, it is not being given in the link. For example, if I am serving this site at http://localhost:8000, the link shows without the ":8000":

```
<entry>
  <title>Willis Library</title>
  <link href="http://localhost/name/nm0000007/" rel="alternate"></link>
  <id>http://localhost/name/nm0000007/</id>
  <summary type="html">Name Type: Building</summary>
</entry>
```
1. In production right now, the Content-Type returned is application/xml and it handles pretty printing in Chrome. Here it is Content-Type: application/atom+xml; charset=utf-8 (which makes sense to me) but  it is not actually pretty printing in Chrome as mentioned at the dev meeting yesterday by @vphill.

### LINKED ISSUE #66: feed entries should have "updated" element
In order to be valid Atom, I believe each entry in /name/feed/ needs to have an "updated" element.

### PR DESCRIPTION
**Pull request by damonkelley**
Will resolve #42 

It would be great if we could approve and merge #49 beforehand so that I can add a section on configuration to the docs to cover the new app level settings that are added in the PR. 

Edit: As of 2462d7d, this will also close #66.

### CODE REVIEW

**Comment by damonkelley** [comment]
And this is what the feed looks like as of 7585ba5

``` xml
<feed xmlns="http://www.w3.org/2005/Atom" xmlns:georss="http://www.georss.org/georss" xml:lang="en-us">
    <title>Name App</title>
    <link href="http://localhost:8000/name/feed/" rel="alternate"/>
    <link href="http://localhost:8000/name/feed/" rel="self"/>
    <id>http://localhost:8000/name/feed/</id>
    <updated>2015-05-18T14:08:54Z</updated>
    <author>
        <name>Administrator</name>
        <email>admin@example.com</email>
        <uri>http://example.com</uri>
    </author>
    <subtitle>New Name Records</subtitle>
    <entry>
        <title>University of North Texas</title>
        <link href="http://localhost:8000/name/nm0000007/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000007/</id>
        <summary type="html">Name Type: Building</summary>
        <georss:point>33.2074880000, -97.1525862000</georss:point>
    </entry>
    <entry>
        <title>Test</title>
        <link href="http://localhost:8000/name/nm0000006/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000006/</id>
        <summary type="html">Name Type: Personal</summary>
    </entry>
    <entry>
        <title>Test</title>
        <link href="http://localhost:8000/name/nm0000005/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000005/</id>
        <summary type="html">Name Type: Personal</summary>
    </entry>
    <entry>
        <title>Test</title>
        <link href="http://localhost:8000/name/nm0000004/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000004/</id>
        <summary type="html">Name Type: Personal</summary>
    </entry>
    <entry>
        <title>Colorful</title>
        <link href="http://localhost:8000/name/nm0000003/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000003/</id>
        <summary type="html">Name Type: Organization</summary>
    </entry>
    <entry>
        <title>Damon Kelley</title>
        <link href="http://localhost:8000/name/nm0000002/" rel="alternate"/>
        <id>http://localhost:8000/name/nm0000002/</id>
        <summary type="html">Name Type: Personal</summary>
        <georss:point>32.8337130000, -96.7504120000</georss:point>
    </entry>
</feed>

```

~~EDIT: Note that I did not have the `NAME_FEED_AUTHOR_*` settings set when I grab this feed to the feed level author elements are not present.~~

File: name/feeds.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    def root_attributes(self):
+        attrs = super(NameAtomFeedType, self).root_attributes()
+        attrs['xmlns:georss'] = 'http://www.georss.org/georss'
+        return attrs
+
+    def add_item_elements(self, handler, item):
+        super(NameAtomFeedType, self).add_item_elements(handler, item)
+        if item.get('location'):
+            handler.addQuickElement('georss:point', item['location'])
```
Comment (damonkelley) [inline name/feeds.py]: Change this to access the `item` dict in a consistent way.

**Comment by damonkelley** [comment]
ping @ldko @somexpert 

P.S. @ldko, I read through my comments and docstrings a little closer this time and I may have caught all of the typos :clap:

**Comment by damonkelley** [comment]
cancel that ping. I want to update the docs first.

**Comment by damonkelley** [comment]
re-ping @ldko @somexpert

**Comment by ldko** [comment]
@vphill should we configure in default information for the feed's author element (since without it a feed is not valid Atom), or is it preferred just to document it as it is (an author element won't appear if not set by the user) and have the user responsible for adding an author to make the feed valid?

**Comment by damonkelley** [comment]
> or is it preferred just to document it as it is (an author element won't appear if not set by the user) and have the user responsible for adding an author to make the feed valid

And if we choose this path, perhaps it would be a good idea to add a warning to the docs that if those settings are not set, then the Atom feed will not be valid.

P.S. @vphill, here is the link to the documentation @ldko mentioned. http://django-name.readthedocs.org/en/update-feed/configuration/#settings

File: name/feeds.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        attrs = super(NameAtomFeedType, self).root_attributes()
+        attrs['xmlns:georss'] = self.GEO_RSS_NS
+        return attrs
+
+    def add_item_elements(self, handler, item):
+        super(NameAtomFeedType, self).add_item_elements(handler, item)
+
+        # Add the georss:point element if the item has a geo_point
+        # attribute.
+        geo_point = item.get('geo_point', None)
```
Comment (ldko) [inline name/feeds.py:29]: I think we should change this to take the comma out of the geo_point http://www.georss.org/ @vphill @damonkelley

Comment (damonkelley) [inline name/feeds.py:29]: Yup, i agree. We can change that [here](https://github.com/unt-libraries/django-name/blob/2428de0dfeb3758b7b74fcadf50466948b4127dc/name/models.py#L544-L545).

**Comment by ldko** [comment]
per Mark, we will set a default author name to the default name of the django-name app, so I guess that would be django-name...

People can use the settings you added if they want to change it or put in an email or uri.

**Comment by damonkelley** [comment]
That works for me.

And the  `author_email` and `author_link` will continue to default to `None`?

**Comment by ldko** [comment]
Yeah, `author_email` and `author_link` can default to `None`. Only `name` should be required to make it valid.

**Comment by ldko** [comment]
Perhaps modify the documentation to reflect this

**Comment by damonkelley** [comment]
Absolutely!

File: name/feeds.py (hunk 2 above)
Comment (damonkelley) [inline name/feeds.py:29]: See d47e276

**Comment by damonkelley** [comment]
Ok, this is where we are now
- The`author_name` now defaults to `"Django Name"`
  - The docs have been updated for that
- The entry `<update>` element now displays the items `last_modified` field.
- The `geo_point` no longer has a comma separating the lng and lat.

I want to point out that there is still a note in the docs in the configuration section that reads as follows

> All feed settings are optional. If you choose not to use them, the author tree will simply not be present in the feed.

Now that there is a default value for `NAME_FEED_AUTHOR_NAME`, the author section will always be present unless the user assigns that constant to `None`.  Therefore, the note is still relevant, but perhaps not worded correctly.

**Comment by ldko** [comment]
Then maybe something like: "All feed settings are optional. If you choose not to use them, the author tree will simply not be present in the feed. However, without an `author` element containing a `name` sub-element, the feed will not successfully validate against the Atom format."

**Comment by damonkelley** [comment]
@ldko, what do you think about this http://django-name.readthedocs.org/en/update-feed/configuration.html

**Comment by damonkelley** [comment]
I am starting to think that we probably don't need to publish a warning about the validity of the Atom feed.

I think that a note stating that the all the feed settings are optional is just fine. Which really meant to imply...

> "If you decide not to use these, your project isn't going to crash."

**Comment by ldko** [comment]
@damonkelley we could get rid of the warning and just stick a "For the feed to be valid according to the Atom specification, an `<author/>` element containing a `<name/>` element is required." in with the note

**Comment by ldko** [comment]
Warnings scare people!

**Comment by damonkelley** [comment]
I like that better

**Comment by damonkelley** [comment]
That should do it.

**Comment by ldko** [comment]
:+1:

**Pull request merged**